os/fs/smartfs/smartfs_procfs.c

```diff
@@ -928,6 +937,54 @@ static size_t smartfs_erasemap_read(FAR struct file *filep, FAR char *buffer, si
 }
 #endif
 
+/****************************************************************************
+ * Name: smartfs_dump_all_sector
+ *
+ * Description: Performs the dump of all sector data
+ *
+ ****************************************************************************/
+
+static ssize_t smartfs_dump_all_sector(FAR struct file *filep, FAR char *buffer, size_t buflen)
+{
+	struct mtd_smart_debug_data_s debug_data;
+	FAR struct smartfs_file_s *priv;
+
+	priv = (FAR struct smartfs_file_s *)filep->f_priv;
+
+	/* Populate the debug_data structure */
+
+	debug_data.debugcmd = SMART_DEBUG_CMD_DUMP_SECTOR;
+	debug_data.debugdata = (uint32_t)SMART_DEBUG_DUMP_ALL;
+
+	priv->level1.mount->fs_blkdriver->u.i_bops->ioctl(priv->level1.mount->fs_blkdriver, BIOC_DEBUGCMD, (unsigned long)&debug_data);
+
+	return buflen;
+}
+
+/****************************************************************************
+ * Name: smartfs_dump_sector
+ *
+ * Description: Performs the dump of specific sector data
+ *
+ ****************************************************************************/
+
+static ssize_t smartfs_dump_sector(FAR struct file *filep, FAR const char *buffer, size_t buflen)
+{
+	struct mtd_smart_debug_data_s debug_data;
+	FAR struct smartfs_file_s *priv;
+
+	priv = (FAR struct smartfs_file_s *)filep->f_priv;
+
+	/* Populate the debug_data structure */
+
+	debug_data.debugcmd = SMART_DEBUG_CMD_DUMP_SECTOR;
+	debug_data.debugdata = atoi(buffer);
+
+	priv->level1.mount->fs_blkdriver->u.i_bops->ioctl(priv->level1.mount->fs_blkdriver, BIOC_DEBUGCMD, (unsigned long)&debug_data);
+
+	return buflen;
```

Comment:
I think maybe it cannot be a problem, but anyway, `buflen` is unsigned, but return type is signed. Is it okay?

Reply:
it doesn't return data, so I changed to return 0;